packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -0,0 +1,39 @@
+import {IReputation} from "../IReputation";
+import {Checkpoint, Epoch, Slot} from "@chainsafe/eth2.0-types";
+
+export function getTargetEpoch(peers: IReputation[], currentCheckPoint: Checkpoint): Epoch {
+  const peersWithHigherFinalizedEpoch = peers.filter(peer => {
+    if(!peer.latestHello) {
+      return false;
+    }
+    if(peer.latestHello.finalizedEpoch > currentCheckPoint.epoch) {
+      return true;
+    }
+  });
+  if(peersWithHigherFinalizedEpoch.length > 0) {
+    return currentCheckPoint.epoch + 1;
+  }
+  return currentCheckPoint.epoch;
+}
+
+export type Chunk = [Slot, Slot];
```

Comment:
> I feel like we should be able to start processing blocks as we receive them, rather than saving them all up until the end.

If you processing them all at once, you have to verify signatures and everything(as some peers might be malicious and sending wrong blocks). Here we collect them and verify if all block roots match (block.parentRoot) from different peers and execute fast state transition without signature and different root verifications (faster state transition). Target is still small (1 epoch) so we will process blocks in batches of 64 or something

> Also I wonder what the tradeoffs with using the `step` param in beacon_block to interleave requests look like, rather than requesting contiguous ranges.

Not sure, I would leave experimentation with this until we have some stable testnet against which we can benchmark syncing